cf/data/data.py

```diff
@@ -7412,45 +7411,45 @@ def apply_masking(
                 fill_values = ()
         else:
             try:
-                _ = iter(fill_values)
+                iter(fill_values)
             except TypeError:
                 raise TypeError(
                     "'fill_values' parameter must be a sequence or "
-                    "of type bool. Got type {}".format(type(fill_values))
+                    f"of type bool. Got type {type(fill_values)}"
                 )
             else:
                 if isinstance(fill_values, str):
                     raise TypeError(
                         "'fill_values' parameter must be a sequence or "
-                        "of type bool. Got type {}".format(type(fill_values))
+                        f"of type bool. Got type {type(fill_values)}"
```

Comment:
Perhaps we can also prevent the duplication of this message whilst we are here, by assigning it to a variable just before the `try/except`?

Reply:
Given that almost all uses will not raise an exception, I'm slightly against creating the message which almost always never get used.

Reply:
Fair enough! I appreciate your justifications. Marking as resolved.